onadata/apps/main/views.py

```diff
@@ -1393,3 +1416,19 @@ def username_list(request):
         data = [user['username'] for user in users]
 
     return HttpResponse(json.dumps(data), content_type='application/json')
+
+def _make_authenticated_request(user):
+    """
+    This is a hack to allow for kobocat to make authenticated requests to KPI's
+    migrate endpoint.
+    Returns response from KPI.
+    """
+    token_key, created = Token.objects.get_or_create(user=user)
```

Comment:
I think that the `Token` sync is one way from KPI to KoBoCAT, meaning that if we add a token in KPI, it should appear in KoBoCAT's database, but not the other way around. Sessions are shared between KPI and KoBoCAT, though—they're stored in a shared Redis database—so maybe we could authenticate that way?